pkg/k2/constructs/construct_marshaller_test.go

```diff
@@ -535,10 +521,12 @@ func TestConstructMarshaller_marshalRefs(t *testing.T) {
 				ConstructEvaluator: evaluator,
 			}
 			got, err := marshaller.marshalRefs(tt.args.o, tt.args.rawVal)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("ConstructMarshaller.marshalRefs() error = %v, wantErr %v", err, tt.wantErr)
-				return
+			if tt.wantErr && assert.Error(t, err, "ConstructMarshaller.marshalRefs() error = %v, wantErr %v", err, tt.wantErr) {
+				t.FailNow()
+			} else if !tt.wantErr && !assert.NoError(t, err, "ConstructMarshaller.marshalRefs() error = %v, wantErr %v", err, tt.wantErr) {
+				t.FailNow()
 			}
```

Comment:
```suggestion
			if tt.wantErr  {
				require.Error(t, err)
			} else {
				require.NoError(t, err)
			}
```
nit: using `require` gets the FailNow behaviour built in, and the message seems redundant (assert/require already print the error if fails and the test name already has `marshalRefs` in it)